### PR TITLE
Some UT for cloud

### PR DESCRIPTION
**Pull request by isbm**
### What does this PR do?

Adds/fixes UT for https://github.com/openSUSE/salt/commit/f829d6f9fc4ffa97b2c175a4e915a174631eba72 that are already here accepted, but untested. :-)

### Tests written?

Yes

### CODE REVIEW

**Comment by gtmanfred** [comment]
Can you actually nuke the rackspace drivers?

The rackspace firstgen cloud was officially deprecated finally last month, so I am going through and removing references to it today.

Thanks,
Daniel

**Comment by gtmanfred** [comment]
If you wouldn't mind removing the rackspace tests.  I am nuking that whole module over here https://github.com/saltstack/salt/pull/40820 because the rackspace firstgen cloud has been deprecated.

Thanks,
Daniel

**Comment by isbm** [comment]
@gtmanfred bah! Another reason to write UT... Sure, less code = less bugs. :laughing:

**Comment by isbm** [comment]
@gtmanfred Here, you have it.

**Comment by gtmanfred** [comment]
Awesome! thanks!

File: salt/cloud/clouds/dimensiondata.py [hunk 1]
```diff
@@ -131,6 +131,67 @@ def get_dependencies():
...
+        data.private_ips = private
+        if ssh_interface(vm_) == 'private_ips':
+            return data
+
+    if public:
+        data.public_ips = public
+        if ssh_interface(vm_) != 'private_ips':
+            return data
+
+    log.debug('DATA')
```
Comment (cachedout) [inline salt/cloud/clouds/dimensiondata.py]: Can we either make these more descriptive or remove these debug statements entirely?

File: salt/cloud/clouds/nova.py [hunk 2]
```diff
@@ -722,6 +722,159 @@ def request_instance(vm_=None, call=None):
...
+            private_ip = preferred_ip(vm_, [private_ip])
+            if private_ip is False:
+                continue
+            if salt.utils.cloud.is_public_ip(private_ip):
+                log.warning('{0} is a public IP'.format(private_ip))
+                data.public_ips.append(private_ip)
+                log.warning(
+                    (
+                        'Public IP address was not ready when we last'
+                        ' checked.  Appending public IP address now.'
```
Comment (cachedout) [inline salt/cloud/clouds/nova.py]: Extra space between sentences here.

Comment (isbm) [inline salt/cloud/clouds/nova.py]: @cachedout Actually extra parenthesis here is more disturbing, yet that wasn't my code. But I will fix that too. :)

Comment (cachedout) [inline salt/cloud/clouds/nova.py]: Heh, yes. Good call.

**Comment by isbm** [comment]
@cachedout made it all prettier, as you wished :-]

**Comment by cachedout** [comment]
Thanks as always, @isbm :]

**Pull request merged**